src/main/java/de/picturesafe/search/elasticsearch/model/SearchResultItem.java

```diff
@@ -30,33 +30,51 @@
  */
 public class SearchResultItem {
 
-    private final long id;
+    private final String id;
     private final Map<String, Object> attributes;
+    private final IdFormat idFormat;
 
     /**
      * Constructor
      *
-     * @param attributes            Attributes of the result item's document
+     * @param attributes    Attributes of the result item's document
      */
     public SearchResultItem(Map<String, Object> attributes) {
-        this.id = extractId(attributes);
-        this.attributes = attributes;
+        this(attributes, IdFormat.DEFAULT);
     }
 
-    private long extractId(Map<String, Object> attributes) {
-        final Long id = ElasticDocumentUtils.getId(attributes);
-        return (id != null) ? id : -1;
+    /**
+     * Constructor
+     *
+     * @param attributes    Attributes of the result item's document
+     * @param idFormat      {@link IdFormat}
+     */
+    public SearchResultItem(Map<String, Object> attributes, IdFormat idFormat) {
+        this.id = ElasticDocumentUtils.getId(attributes);
+        this.attributes = attributes;
+        this.idFormat = idFormat;
     }
 
     /**
      * Gets the ID of the result items's document.
      *
      * @return ID
      */
-    public long getId() {
+    public String getId() {
         return id;
     }
 
+    /**
+     * Gets the ID of the result items's document.
+     *
+     * @param <T> Type of the ID
```

Comment:
Minor: Javadoc format

Reply:
Please see also Javadoc of getAttribute(String name)

Reply:
What is the promblem?